worktree.go

```diff
@@ -381,6 +434,20 @@ func (w *Worktree) resetWorktree(t *object.Tree) error {
 	b := newIndexBuilder(idx)
 
 	for _, ch := range changes {
+		if len(files) > 0 {
+			file := ""
+			if ch.From != nil {
+				file = ch.From.Name()
+			} else {
+				file = ch.To.Name()
+			}
+
```

Comment:
I am not sure we correctly validate for either `ch.From` and `ch.To` to be present. To avoid a panic in that case, we may want to check here and `continue` if `file` is empty.

```suggestion
			} else if ch.To != nil {
				file = ch.To.Name()
			}
			
			if file == "" {
				continue
			}
```